### PR TITLE
build kube-discovery and kubeadm with release

### PR DESCRIPTION
**Pull request by mikedanese**
This will start publishing binaries from our ci builds.

@errordeveloper @dgoodwin

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33496)

### CODE REVIEW

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -38,6 +38,8 @@ kube::golang::server_targets() {
     cmd/kubeadm
     cmd/kubemark
     cmd/hyperkube
+    cmd/kubeadm
```
Comment (luxas) [inline hack/lib/golang.sh]: I added kubeadm above already :)

**Comment by luxas** [comment]
@ixdy @spxtr @fejta How far have you come with `make release`?
We need it here...

File: hack/lib/golang.sh (hunk 1 above)
Comment (mikedanese) [inline hack/lib/golang.sh]: Oops. ok fixed.

**Comment by ixdy** [comment]
@k8s-bot build this

**Comment by ixdy** [comment]
@luxas now available on request :)

**Comment by luxas** [comment]
@ixdy Great, thanks a lot!
Will the build upload the artifacts somewhere?
That's what we'd like for our @kubernetes/sig-cluster-lifecycle work...

**Comment by ixdy** [comment]
> Will the build upload the artifacts somewhere?

PR Jenkins does not. The CI cross-build does not upload anywhere currently, either.

**Comment by luxas** [comment]
Hmm, what were you thinking then @mikedanese?
Would you hook up some external jenkins job or...?

**Comment by ixdy** [comment]
We can fairly easily upload the results from CI crossbuilds; we'd just need to be careful not to collide with the artifacts from the CI quick-release build. We hadn't bothered setting that up yet, since nobody was consuming the CI build results from anything other than linux/amd64.

**Comment by luxas** [comment]
Now we are :)

**Comment by ixdy** [comment]
@luxas @mikedanese can one of you open an issue in kubernetes/test-infra explaining what you're doing with the release artifacts so we can fully understand what you need?

**Comment by luxas** [comment]
@mikedanese ^

**Comment by luxas** [comment]
But @mikedanese I do not understand why we'd like to build kube-discovery normally.
It should exist only as an image for the time being, we do not need it downloadable AFAIU

Please update

**Comment by mikedanese** [comment]
@luxas As long as this is a standalone in cmd/, we should publish this as a server binary like we publish kube-controller-manager, kube-apiserver, kube-dns, etc... This should be available later in the pipeline and we should be testing the compilation of these binaries. Right now we are not testing whether this compiles during CI.

**Comment by jbeda** [comment]
+1 If you aren't building and testing it'll break.

**Comment by luxas** [comment]
OK, using it for validation SGTM

I think this is OK then.

**Pull request merged**